pages/api/teams/[teamId]/documents/index.ts

```diff
@@ -84,7 +86,14 @@ export default async function handle(
       // Get passed type property or alternatively, the file extension and save it as the type
       const type = fileType || getExtension(name);
 
-      // You could perform some validation here
+      // Check whether the Notion page is publically accessible or not
+      try {
+        const pageId = parsePageId(fileUrl, { uuid: false });
+        // if the page isn't accessible then end the process here.
+        await notion.getPage(pageId);
+      } catch (error) {
+        return res.status(404).end("This Notion page isn't publically available.");
+      }
```

Comment:
Check if the `type === "notion"` otherwise, this will throw an error for regular documents as the backend function is shared

Reply:
Sounds good 👍